service/history/replication/batchable_task.go

```diff
@@ -0,0 +1,127 @@
+package replication
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"go.temporal.io/server/common/backoff"
+	ctasks "go.temporal.io/server/common/tasks"
+)
+
+type (
+	BatchableTask interface {
+		TrackableExecutableTask
+		Combine(task BatchableTask) (BatchableTask, error)
+	}
+
+	// batchedTask should be only used by sequential_combined_queue
+	batchedTask struct {
+		batchedTask     BatchableTask
+		individualTasks []BatchableTask
+		lock            sync.Mutex
+		state           batchState
+	}
+
+	batchState int
+)
+
+const (
+	batchStateOpen  = 0
+	batchStateClose = 1
+)
+
+var _ TrackableExecutableTask = (*batchedTask)(nil)
+
+func (w *batchedTask) QueueID() interface{} {
+	return w.batchedTask.QueueID()
+}
+
+func (w *batchedTask) TaskID() int64 {
+	return w.individualTasks[0].TaskID()
+}
+
+func (w *batchedTask) TaskCreationTime() time.Time {
+	return w.individualTasks[0].TaskCreationTime()
+}
+
+func (w *batchedTask) MarkPoisonPill() error {
+	if len(w.individualTasks) == 1 {
+		return w.batchedTask.MarkPoisonPill()
+	} else {
+		return nil
```

Comment:
this should not return nil, but rather attempting to break the batched task and apply one by one?
plz add a TODO here

Reply:
or `len(w.individualTasks) > 1` is not expected? then logic should panic here